src/main/java/com/microsoft/cse/helium/app/controllers/FeaturedController.java

```diff
@@ -26,13 +32,20 @@
       produces = MediaType.APPLICATION_JSON_VALUE)
   public Mono<Movie> getFeaturedMovies() {
 
-    return featuredMovieDao
-        .getFeaturedMovie()
-        .collectList()
-        .flatMap(
-            featuredMovies -> {
-              int randomNum = new Random().nextInt(((featuredMovies.size() - 1) - 0) + 1) + 0;
-              return moviesDao.getMovieById(featuredMovies.get(randomNum));
-            });
+    try {
+      return featuredMovieDao
+          .getFeaturedMovie()
+          .collectList()
+          .flatMap(
+              featuredMovies -> {
+                int randomNum = new Random().nextInt(((featuredMovies.size() - 1) - 0) + 1) + 0;
+                return moviesDao.getMovieById(featuredMovies.get(randomNum));
+              });
+    } catch (Exception ex) {
```

Comment:
WRT to the mixing of Exception and error codes, this is a good example of appropriate exception handling. We cannot propagate the exception anymore, we we handle it here and just print what we can and send back the data in appropriate format to the caller.

Reply:
Thanks :)